Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/URLSessionSwizzlerTests.swift

```diff
@@ -17,24 +17,30 @@ extension URLSessionSwizzler {
 }
 
 class URLSessionSwizzlerTests: XCTestCase {
-    private let interceptor = URLSessionInterceptorMock()
+    private let interceptor = URLSessionInterceptorMock.mockAny()
     private var swizzler: URLSessionSwizzler! // swiftlint:disable:this implicitly_unwrapped_optional
 
+    private var currentAutoInstrumentation: URLSessionAutoInstrumentation?
     override func setUpWithError() throws {
-        super.setUp()
+        try super.setUpWithError()
         swizzler = try URLSessionSwizzler()
         swizzler.swizzle()
+
+        currentAutoInstrumentation = URLSessionAutoInstrumentation.instance
+        URLSessionAutoInstrumentation.instance = URLSessionAutoInstrumentation(interceptor: interceptor)
     }
 
     override func tearDown() {
         swizzler.unswizzle()
+
+        URLSessionAutoInstrumentation.instance = currentAutoInstrumentation
```

Comment:
Same: we shouldn't need to store the previous instance (`currentAutoInstrumentation`). Each test should start with a clean state and reset it after completion, otherwise it will introduce flakiness. This should be enough:
```swift
URLSessionAutoInstrumentation.instance = nil
```